Added sphinx documentation

raoashish10 opened the pull request:
<!-- Please create/claim an issue before sending a PR -->
<!-- Add issue number (Eg: fixes #123) -->

### Checklist
- [x] My branch is up-to-date with upstream/develop branch.
- [x] Everything works and tested for Python 3.5.2 and above.

### Description
<!-- Describe about what this PR does, previous state and new state of the output -->
Added a sphinx documentation for the hydra-python-agent repo as I noticed that it didn't have one. Also the documentation can then be hosted on readthedocs so that it can be embedded into the new documentation. The sphinx documentation was created specially for the automodule feature
### Change logs

#### Added 
A docs folder which contains the sphinx documentation. 
<!-- Edit these points below to describe the new features added with this PR -->
<!-- - Feature 1 -->
<!-- - Feature 2 -->


<!-- #### Changed -->
<!-- Edit these points below to describe the changes made in existing functionality with this PR -->
<!-- - Change 1 -->
<!-- - Change 1 -->


<!-- #### Fixed -->
<!-- Edit these points below to describe the bug fixes made with this PR -->
<!-- - Bug 1 -->


<!-- #### Removed -->
<!-- Edit these points below to describe the removed features with this PR -->
<!-- - Deprecated feature 1 -->

I am running into some errors while using the agent with the new Movie API doc. So the index page is incomplete because it has certain examples missing which I will add as soon as I fix the erros.

raoashish10 committed:
Added sphinx documentation

raoashish10 committed:
Corrected syntax

priyanshunayan commented:
Can you let me know what are errors that you are facing while using the agent so that we can fix them?

priyanshunayan reviewed (commented):
Adding the steps to run the sphinx docs locally can be a good addition in the readme?

raoashish10 commented:
@priyanshunayan Sure! I will add the steps. My doubt was that when I am trying to run the put function with the new Apidoc, I am getting an error: 
![image](https://user-images.githubusercontent.com/54390491/107187009-4d5c2100-6a0b-11eb-86f3-555d82936b54.png)

Am I missing any steps or am I making any mistake?

raoashish10 committed:
Added steps to run docs locally

raoashish10 committed:
Update README.md

raoashish10 committed:
Fixed and added new examples

raoashish10 commented:
@priyanshunayan I managed to fix the error (@farazkhanfk7 helped out) and I have added examples for different types of request with the Movie api doc. Let me know if anything else needs to be changed. I have also added the steps to build the docs locally in a README file.

priyanshunayan commented on docs/README.md:
````diff
@@ -0,0 +1,18 @@
+# How to  run the docs locally
+
+1. <b>(Skip this step if you have already installed hydrus)</b> <br>
+&nbsp;i) Clone the hydrus repository: ```git clone https://github.com/HTTP-APIs/hydrus.git```<br>
````
Why are we installing hydrus to build the docs?

raoashish10 replied:
I am sorry this is a blunder on my side I will remove the hydrus installation part. I must not have paid proper attention

priyanshunayan commented on docs/source/conf.py:
```diff
@@ -0,0 +1,55 @@
+
+project = 'hydra-python-agent'
+copyright = '2021, Ashish Rao'
+author = 'Ashish Rao'
```
author to Hydra Ecosystem, maybe?

raoashish10 replied:
Sure that would be better!

Mec-iS replied:
Everything contributed to Hydra Ecosystem is under MIT Licence and copyright has to be assigned to "Hydra Ecosystem" or "collective@hydraecosystem.org"

priyanshunayan commented on docs/source/conf.py:
```diff
@@ -0,0 +1,55 @@
+# The theme to use for HTML and HTML Help pages.  See the documentation for
+# a list of builtin themes.
+#
+html_theme = 'sphinx_rtd_theme'
```
`sphinx_rtd_theme` in the requirements.txt file as well as it is no longer a hard dependency in the latest versions.

raoashish10 replied:
Sure I will add that!

priyanshunayan reviewed (dismissed):
Overall a ✨ work! 

I have suggested a few minor changes and we would be good to go!

raoashish10 committed:
Added requirements and made changes to files
